eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -133,66 +131,56 @@ stages:
-                          git status
-
-                          # Commit changes
-                          Write-Host "git add -A"
-                          git add -A 
-                          Write-Host "git -c user.name=`"azure-sdk`" -c user.email=`"azuresdk@microsoft.com`" commit -m `"Initial vcpkg commit for ${{ artifact.VcpkgPortName }}`""
-                          git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" commit  -m "Initial vcpkg commit for ${{ artifact.VcpkgPortName }}" 
-
-                          # Run vcpkg x-add-version to add version to vcpkg file
-                          Write-Host "./bootstrap-vcpkg.bat"
-                          ./bootstrap-vcpkg.bat
-                          Write-Host "./vcpkg.exe x-add-version ${{ artifact.VcpkgPortName }}"
-                          ./vcpkg.exe x-add-version ${{ artifact.VcpkgPortName }}
-
-                          # Amend commit to include results of x-add-version
-                          Write-Host "git status"
+                      # Check out the PR branch if it's already in remote. 
+                      # Ignore failures.
+                      - pwsh: |
+                          $ErrorActionPreference = "Continue"
+                          git checkout "origin/$(PrBranchName)" 2>&1 | Out-Null
+                          $LASTEXITCODE = 0 # This ignores any error from git checkout
                           git status
-                          Write-Host "git add -A"
-                          git add -A
-                          Write-Host "git -c user.name=`"azure-sdk`" -c user.email=`"azuresdk@microsoft.com`" commit --amend -m `"[${{ artifact.VcpkgPortName }}] Update to $(PackageVersion)`""
-                          git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" commit --amend -m "[${{ artifact.VcpkgPortName }}] Update to $(PackageVersion)"
-
-                          # Work around create-pull-request.yml logic that 
-                          # checks for changes
-                          Write-Host "##vso[task.setvariable variable=HasChanges]$true"
-                        displayName: Commit changes for PR
+                        displayName: Checkout Previous PRBranch if it exist.
+                        workingDirectory: $(VcpkgWorkspace)
 
+                      - task: Powershell@2
+                        inputs:
+                          pwsh: true
+                          targetType: filePath
+                          filePath: eng/scripts/Update-VcpkgPort.ps1
+                          arguments: >-
+                            -ReleaseArtifactSourceDirectory "$(Pipeline.Workspace)/packages/${{ artifact.Name }}"
+                            -PortDestinationDirectory 'ports/${{ artifact.VcpkgPortName }}'
+                            -VcpkgPortName '${{ artifact.VcpkgPortName }}'
+                            -GitCommitParameters '-c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com"'
```

Comment:
Are the commit parameters interesting given that you are going to revert it?

Reply:
If we don't supply these paramters on the first commit I think git will complain that they need user.name and user.email, however we do need them on the second commit because we use `commit --file` instead of `commit -m` to put a multiline commit message in.